src/Control/Carrier/NonDet/Maybe.hs

```diff
@@ -1,4 +1,9 @@
 {-# LANGUAGE FlexibleInstances, GeneralizedNewtypeDeriving, MultiParamTypeClasses, TypeOperators, UndecidableInstances #-}
+{- | A carrier for 'NonDet' effects providing choice and failure.
+
+Unlike the Church-encoded 'NonDet' carrier, this carrier terminates immediately upon finding a successful result.
```

Comment:
Again, it isn’t true that `NonDet.Church` doesn’t terminate; it can express exactly the same behaviour as this carrier.

Reply:
Ah, I misunderstood!